RevenueCatUI/CustomerCenter/Data/SubscriptionInformation.swift

```diff
@@ -14,33 +14,178 @@
 //
 
 import Foundation
+import RevenueCat
 
+// swiftlint:disable nesting
 struct SubscriptionInformation {
 
-    let title: String
-    let durationTitle: String
-    let price: String
-    let expirationDateString: String?
+    let title: String?
+    let explanation: Explanation?
+    let durationTitle: String?
+    let price: PriceDetails
+    let expirationOrRenewal: ExpirationOrRenewal?
     let productIdentifier: String
 
     let willRenew: Bool
     let active: Bool
 
-    init(title: String,
-         durationTitle: String,
-         price: String,
-         expirationDateString: String?,
+    let store: Store?
+
+    init(title: String?,
+         explanation: Explanation?,
+         durationTitle: String?,
```

Comment:
What does it mean if these are all nil? Are there any constraints or is every combination of nil possible? (E.g. only the title is nil, both the title and explanation are nil, only the explanation is nil, etc.)

If there are constraints, maybe we can make constructors that force those constraints?

Reply:
that's a good point. I will write previews and see if there are some common constraints